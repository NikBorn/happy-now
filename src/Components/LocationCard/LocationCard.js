import React, { Component } from 'react';
import { toggleFavorite, 
  removeFavorite, 
  addFavorite, 
  toggleExtended } from '../../actions';
import { switchFavorite, switchExtended } from '../../Utils/helper.js';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import HappyHourForm from '../HappyHourForm/HappyHourForm.js';
import fire from '../../fire.js';

const LocationCard = (props) => {

  const locationInfo = props.locationInfo;
  
  const locationDirections = `https://maps.google.com/?q=
    ${locationInfo.location.lat},${locationInfo.location.lng}`;

  const cardStyle = locationInfo.isFavorite ? 
    'favorite-card-header card-header' 
    : 
    'card-header';

  const favStyle = locationInfo.isFavorite ? 
    'favorite-button favorite-button-selected' 
    : 
    'favorite-button';

  const handleClick = (location) => {
    if (location.isFavorite) {
      props.removeFavorite(location);
    } else {
      props.addFavorite(location);
    }
  };

  const locationMenu = locationInfo.menu ? 
    <h5 className='menu'>
      <a href={locationInfo.menu.url} className='directions'>
      menu
      </a>
    </h5> 
    : 
    <h5 className='menu'>
      <a href='' className='directions'>
        no menu listed
      </a>
    </h5>;

  const cardExtStyle = locationInfo.isExtended ? 
    'location-card-ext location-card' 
    : 
    'location-card';

  const addFavToFirebase = () => {
    const itemsRef = fire.database().ref('favorites');
    const item = {
      userId: props.activeUser.uid,
      userName: props.activeUser.displayName,
      location: props.locationInfo
    };
    itemsRef.push(item);
  };

  return (
    <div className={cardExtStyle}
    >
      <div className={cardStyle}>
        <h4>{locationInfo.name}</h4>
        <button className={favStyle}
          onClick={(event) => {
            event.preventDefault();
            handleClick(locationInfo);
            addFavToFirebase();
            props.toggleFavorite(switchFavorite(locationInfo));
          }}>{locationInfo.isFavorite ? 'Unfav' : 'Fav'}</button>
      </div>
      <h5>{locationInfo.contact.formattedPhone || 'No Phone Number Listed'}</h5>
      <h5>{locationInfo.location.formattedAddress[0]}</h5>
      <h5>{locationInfo.location.formattedAddress[1]}</h5>
      <h5>{locationInfo.rating} rating from {locationInfo.ratingSignals} reviews
      </h5>
      {locationMenu}
      <h5 className='directions'>
        <a href={locationDirections}
          className='directions'
          target="_blank">
        directions
        </a>
      </h5>
      {
        locationInfo.isExtended === false &&
        <button
          className='show-more happy-hour-btn'
          onClick={event => {
            event.preventDefault();
            props.toggleExtended(switchExtended(locationInfo));
          }}
        >Add Happy Hour!</button>
      }
      <HappyHourForm locationInfo={locationInfo} />
    </div>
  );
  
};


LocationCard.propTypes = {
  locationInfo: PropTypes.object,
  removeFavorite: PropTypes.func,
  addFavorite: PropTypes.func,
  toggleFavorite: PropTypes.func,
  toggleExtended: PropTypes.func
};

const mapStateToProps = (state) => {
  return {
    locations: state.locations,
    favorites: state.favorites,
    activeUser: state.activeUser
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    toggleFavorite: (location) => {
      dispatch(toggleFavorite(location));
    },
    addFavorite: (location) => {
      dispatch(addFavorite(location));
    },
    removeFavorite: (location) => {
      dispatch(removeFavorite(location));
    },
    toggleExtended: (location) => {
      dispatch(toggleExtended(location));
    }
  };

};

export default connect(mapStateToProps, mapDispatchToProps)(LocationCard);