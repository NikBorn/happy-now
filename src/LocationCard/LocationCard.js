import React from 'react';

const LocationCard = ({locationInfo}) => {
  console.log(locationInfo)

  return (
    <div>
      <h4>{locationInfo.name}</h4>
      <h6>{locationInfo.contact.formattedPhone}</h6>
      <h6>{locationInfo.rating}/{locationInfo.ratingSignals}</h6>
      <button>Fav</button>
    </div>
  )
}

export default LocationCard;