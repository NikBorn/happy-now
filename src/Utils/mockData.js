

export const mockLocationsResponse = [
  {
    isFavorite: false,
    id: "4b044f71f964a5202e5322e3",
    name: "The Glenn",
    contact: {
      phone: "3032551400",
      formattedPhone: "(303) 255-1400",
      twitter: "biggyglenn",
      facebook: "115875251829275",
      facebookName: "Glenn Bar"},
    location: {
      address: "11140 Irma Dr",
      lat: 39.89855629220959,
      lng: -104.96589290871326,
      labeledLatLngs: [{
        label: "display",
        lat: 39.89855629220959,
        lng: -104.96589290871326
      }],
      distance: 4876,
      postalCode: "80233",
      cc: "US",
      city: "Northglenn",
      state: "CO",
      country: "United States",
      formattedAddress: ['11140 Irma Dr', 'Northglenn, CO 80233', 'United States']},
    categories: [{
      id: "4bf58dd8d48988d116941735",
      name: "Bar",
      pluralName: "Bars",
      shortName: "Bar",
      icon: {
        prefix: "https://ss3.4sqi.net/img/categories_v2/nightlife/pub_",
        suffix: ".png"
      },
      primary: true}],
    verified: false,
    stats: {
      checkinsCount: 2080,
      usersCount: 702,
      tipCount: 29},
    url: "http://theglennbarandgrill.com",
    price: {
      tier: 2,
      message: "Moderate",
      currency: "$"},
    hasMenu: true,
    rating: 8.4,
    ratingSignals: 84,
    menu: { 
      type: "Menu", 
      label: "Menu", 
      anchor: "View Menu",
      url: "https://foursquare.com/v/the-glenn/4b044f71f964a5202e5322e3/menu",
      mobileUrl: "https://foursquare.com/v/4b044f71f964a5202e5322e3/device_menu"},
    allowMenuUrlEdit: true,
    beenHere: {
      count: 0,
      marked: false,
      lastCheckinExpiredAt: 0},
    hours: {
      status: "Likely open",
      richStatus: {
        entities: [],
        text: "Likely open"},
      isOpen: true,
      isLocalHoliday: false},
    photos: {
      count: 0,
      groups: []},
    hereNow: {
      count: 0,
      summary: "Nobody here",
      groups: [] }},
  {
    name: "Tequila Rock",
    price: { tier: 2, message: "Moderate", currency: "$" },
    rating: 6.4,
    ratingSignals: 399,
    contact: { phone: "3032551400", 
      formattedPhone: "(303) 255-1400", 
      twitter: "biggyglenn", 
      facebook: "115875251829275", 
      facebookName: "Glenn Bar" },
    isFavorite: false,
    id: "4d8a35c976e1236a99721534"
  },
  {
    name: "Paddy's Pub",
    price: { tier: 1, message: "Moderate", currency: "$" },
    rating: 3.4,
    ratingSignals: 107,
    contact: { phone: "3038675309", 
      formattedPhone: "(303) 867-5309", 
      twitter: "biggyglenn", 
      facebook: "115875251829275", 
      facebookName: "Glenn Bar" },
    isFavorite: false,
    id: "4d8a35c976e1236a99725553"
  }
];

