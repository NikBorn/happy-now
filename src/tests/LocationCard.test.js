import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import LocationCard from '../Components/LocationCard/LocationCard.js';
import Adapter from 'enzyme-adapter-react-16';
import { MockLocation } from '../Utils/mockData.js';
import configureStore from 'redux-mock-store';

configure({ adapter: new Adapter() })

describe('LocationCard', () => {
  const mockStore = configureStore();
  const initialState = {
    locations: [],
    userLocation: {},
    activeUser: null,
    count: 10,
    favorites: []
  }
  const store = mockStore(initialState);
  const locationInfo = MockLocation
  const wrapper = shallow(<LocationCard store={store} key={locationInfo.id} locationInfo={locationInfo} />)

  it('should always match the Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})