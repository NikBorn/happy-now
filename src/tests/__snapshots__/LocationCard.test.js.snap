// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationCard should always match the Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(LocationCard)
    locationInfo={
        Object {
            "allowMenuUrlEdit": true,
            "beenHere": Object {
              "count": 0,
              "lastCheckinExpiredAt": 0,
              "marked": false,
            },
            "categories": Array [
              Object {
                "icon": Object {
                  "prefix": "https://ss3.4sqi.net/img/categories_v2/nightlife/pub_",
                  "suffix": ".png",
                },
                "id": "4bf58dd8d48988d116941735",
                "name": "Bar",
                "pluralName": "Bars",
                "primary": true,
                "shortName": "Bar",
              },
            ],
            "contact": Object {
              "facebook": "115875251829234",
              "facebookName": "Paddy's Pub",
              "formattedPhone": "(303) 867-5309",
              "phone": "3032551400",
              "twitter": "paddysmfnpub",
            },
            "hasMenu": true,
            "hereNow": Object {
              "count": 0,
              "groups": Array [],
              "summary": "Nobody here",
            },
            "hours": Object {
              "isLocalHoliday": false,
              "isOpen": true,
              "richStatus": Object {
                "entities": Array [],
                "text": "Likely open",
              },
              "status": "Likely open",
            },
            "id": "4b044f71f964a5202e5322e3",
            "isFavorite": false,
            "location": Object {
              "address": "11140 Irma Dr",
              "cc": "US",
              "city": "Northglenn",
              "country": "United States",
              "distance": 4876,
              "formattedAddress": Array [
                "11140 Irma Dr",
                "Northglenn, CO 80233",
                "United States",
              ],
              "labeledLatLngs": Array [
                Object {
                  "label": "display",
                  "lat": 39.89855629220959,
                  "lng": -104.96589290871326,
                },
              ],
              "lat": 39.89855629220959,
              "lng": -104.96589290871326,
              "postalCode": "80233",
              "state": "CO",
            },
            "menu": Object {
              "anchor": "View Menu",
              "label": "Menu",
              "mobileUrl": "https://foursquare.com/v/4b044f71f964a5202e5322e3/device_menu",
              "type": "Menu",
              "url": "https://foursquare.com/v/the-glenn/4b044f71f964a5202e5322e3/menu",
            },
            "name": "Paddy's Pub",
            "photos": Object {
              "count": 0,
              "groups": Array [],
            },
            "price": Object {
              "currency": "$",
              "message": "Moderate",
              "tier": 2,
            },
            "rating": 8.4,
            "ratingSignals": 84,
            "stats": Object {
              "checkinsCount": 2080,
              "tipCount": 29,
              "usersCount": 702,
            },
            "url": "http://theglennbarandgrill.com",
            "verified": false,
          }
    }
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {
      "addFavorite": [Function],
      "favorites": Array [],
      "locationInfo": Object {
        "allowMenuUrlEdit": true,
        "beenHere": Object {
          "count": 0,
          "lastCheckinExpiredAt": 0,
          "marked": false,
        },
        "categories": Array [
          Object {
            "icon": Object {
              "prefix": "https://ss3.4sqi.net/img/categories_v2/nightlife/pub_",
              "suffix": ".png",
            },
            "id": "4bf58dd8d48988d116941735",
            "name": "Bar",
            "pluralName": "Bars",
            "primary": true,
            "shortName": "Bar",
          },
        ],
        "contact": Object {
          "facebook": "115875251829234",
          "facebookName": "Paddy's Pub",
          "formattedPhone": "(303) 867-5309",
          "phone": "3032551400",
          "twitter": "paddysmfnpub",
        },
        "hasMenu": true,
        "hereNow": Object {
          "count": 0,
          "groups": Array [],
          "summary": "Nobody here",
        },
        "hours": Object {
          "isLocalHoliday": false,
          "isOpen": true,
          "richStatus": Object {
            "entities": Array [],
            "text": "Likely open",
          },
          "status": "Likely open",
        },
        "id": "4b044f71f964a5202e5322e3",
        "isFavorite": false,
        "location": Object {
          "address": "11140 Irma Dr",
          "cc": "US",
          "city": "Northglenn",
          "country": "United States",
          "distance": 4876,
          "formattedAddress": Array [
            "11140 Irma Dr",
            "Northglenn, CO 80233",
            "United States",
          ],
          "labeledLatLngs": Array [
            Object {
              "label": "display",
              "lat": 39.89855629220959,
              "lng": -104.96589290871326,
            },
          ],
          "lat": 39.89855629220959,
          "lng": -104.96589290871326,
          "postalCode": "80233",
          "state": "CO",
        },
        "menu": Object {
          "anchor": "View Menu",
          "label": "Menu",
          "mobileUrl": "https://foursquare.com/v/4b044f71f964a5202e5322e3/device_menu",
          "type": "Menu",
          "url": "https://foursquare.com/v/the-glenn/4b044f71f964a5202e5322e3/menu",
        },
        "name": "Paddy's Pub",
        "photos": Object {
          "count": 0,
          "groups": Array [],
        },
        "price": Object {
          "currency": "$",
          "message": "Moderate",
          "tier": 2,
        },
        "rating": 8.4,
        "ratingSignals": 84,
        "stats": Object {
          "checkinsCount": 2080,
          "tipCount": 29,
          "usersCount": 702,
        },
        "url": "http://theglennbarandgrill.com",
        "verified": false,
      },
      "locations": Array [],
      "removeFavorite": [Function],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "toggleFavorite": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "addFavorite": [Function],
        "favorites": Array [],
        "locationInfo": Object {
          "allowMenuUrlEdit": true,
          "beenHere": Object {
            "count": 0,
            "lastCheckinExpiredAt": 0,
            "marked": false,
          },
          "categories": Array [
            Object {
              "icon": Object {
                "prefix": "https://ss3.4sqi.net/img/categories_v2/nightlife/pub_",
                "suffix": ".png",
              },
              "id": "4bf58dd8d48988d116941735",
              "name": "Bar",
              "pluralName": "Bars",
              "primary": true,
              "shortName": "Bar",
            },
          ],
          "contact": Object {
            "facebook": "115875251829234",
            "facebookName": "Paddy's Pub",
            "formattedPhone": "(303) 867-5309",
            "phone": "3032551400",
            "twitter": "paddysmfnpub",
          },
          "hasMenu": true,
          "hereNow": Object {
            "count": 0,
            "groups": Array [],
            "summary": "Nobody here",
          },
          "hours": Object {
            "isLocalHoliday": false,
            "isOpen": true,
            "richStatus": Object {
              "entities": Array [],
              "text": "Likely open",
            },
            "status": "Likely open",
          },
          "id": "4b044f71f964a5202e5322e3",
          "isFavorite": false,
          "location": Object {
            "address": "11140 Irma Dr",
            "cc": "US",
            "city": "Northglenn",
            "country": "United States",
            "distance": 4876,
            "formattedAddress": Array [
              "11140 Irma Dr",
              "Northglenn, CO 80233",
              "United States",
            ],
            "labeledLatLngs": Array [
              Object {
                "label": "display",
                "lat": 39.89855629220959,
                "lng": -104.96589290871326,
              },
            ],
            "lat": 39.89855629220959,
            "lng": -104.96589290871326,
            "postalCode": "80233",
            "state": "CO",
          },
          "menu": Object {
            "anchor": "View Menu",
            "label": "Menu",
            "mobileUrl": "https://foursquare.com/v/4b044f71f964a5202e5322e3/device_menu",
            "type": "Menu",
            "url": "https://foursquare.com/v/the-glenn/4b044f71f964a5202e5322e3/menu",
          },
          "name": "Paddy's Pub",
          "photos": Object {
            "count": 0,
            "groups": Array [],
          },
          "price": Object {
            "currency": "$",
            "message": "Moderate",
            "tier": 2,
          },
          "rating": 8.4,
          "ratingSignals": 84,
          "stats": Object {
            "checkinsCount": 2080,
            "tipCount": 29,
            "usersCount": 702,
          },
          "url": "http://theglennbarandgrill.com",
          "verified": false,
        },
        "locations": Array [],
        "removeFavorite": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "toggleFavorite": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
